File: src/string/circularstring.dart
class Circularstring {
  bool isCircularSentence(String sentence) {
    int i;
    String x;
    for (i = 0; i < sentence.length; i++) {
      x = sentence[i];
      if (x == ' ') {
        if (sentence[i - 1] != sentence[i + 1]) {
          return false;
        }
      }
      if (sentence[0] != sentence[sentence.length - 1]) return false;
    }
    return true;
  }
}

void main() {
    Circularstring ob = Circularstring();
    String sentence = "Leetcode is cool";
    bool ans = ob.isCircularSentence(sentence);
    print(ans);
  }

File: src/string/circularString.java
// que link: https://leetcode.com/problems/circular-sentence/

public class circularString {
      public boolean isCircularSentence(String sentence) {
            int i;
            char x;

            for (i = 0; i < sentence.length(); i++) {
                  x = sentence.charAt(i);
                  if(x == ' ') {
                        if(sentence.charAt(i - 1) != sentence.charAt(i + 1)) {
                              return false;
                        }
                  }
                  if(sentence.charAt(0) != sentence.charAt(sentence.length() - 1))
                        return false;
            }
            return true;
      }

      public static void main(String[] args) {
            circularString ob = new circularString();
            String sentence = "Leetcode is cool";
            boolean ans = ob.isCircularSentence(sentence);
            System.out.println(ans);
      }
}
